src/lib/utils.ts

```diff
@@ -33,6 +34,78 @@ export const getNodeTextContent = (node: ManuscriptNode) => {
   return textContent
 }
 
+export const getContributorTextContent = (
+  node: ManuscriptNode,
+  oldAttrs: Record<any, any> | undefined
+) => {
+  let textContent = ''
+  textContent += `${node.attrs.bibliographicName.given} ${node.attrs.bibliographicName.family}`
+  if (oldAttrs) {
+    textContent =
+      `${oldAttrs.bibliographicName.given} ${oldAttrs.bibliographicName.family} to ` +
+      textContent
+  }
+  return textContent
+}
+
+export const getAffiliationTextContent = (
+  node: ManuscriptNode,
+  oldAttrs: Record<any, any> | undefined
+) => {
+  let textContent = ''
+  textContent += `${node.attrs.institution}`
+  if (oldAttrs) {
+    textContent = `${oldAttrs.institution} to ` + textContent
+  }
+  return textContent
+}
+
+export const findBibliographyById = (
+  doc: ManuscriptNode,
+  id: string
+): ManuscriptNode | null => {
+  let bibNode: ManuscriptNode | null = null
+
+  doc.descendants((node) => {
+    if (node.type === schema.nodes.bibliography_item && node.attrs.id === id) {
+      bibNode = node
+      return false // stop traversal
+    }
+    if (bibNode) {
+      return false
+    }
+    return true // continue traversal
+  })
+  return bibNode
+}
+
+export const getTextContentFromBibliography = (
+  state: ManuscriptEditorState,
+  id: string
+): string => {
+  const bib = findPluginByKey(state, 'bibliography')?.getState(state)
+  const [meta, bibliography] = bib.provider.makeBibliography()
+  const selectedBib = meta.entry_ids.findIndex(
+    (entry: [string]) => entry[0] == id
+  )
+  const parser = new DOMParser()
+  const textContent = parser.parseFromString(
+    bibliography[selectedBib],
+    'text/html'
+  ).body.textContent
+  return textContent ? textContent : ''
+}
+
+export const getFigureLabel = (
+  state: ManuscriptEditorState,
+  node: ManuscriptNode
+) => {
+  const objectsPlugin = findPluginByKey(state, 'objects')
```

Comment:
While do sometimes refer to plugins directly by string in the body-editor, it would be advisable to use a constant or some other reference imported from body-editor to address the plugin here to avoid bugs if a plugin gets renamed or refactored.

Reply:
done